swarm/network/stream/snapshot_sync_test.go

```diff
@@ -290,7 +295,15 @@ func runSyncTest(chunkCount int, nodeCount int, live bool, history bool) error {
 				return err
 			}
 
-			watchSubscriptionEvents(ctx, id, client, errc)
+			wsDoneC := watchSubscriptionEvents(ctx, id, client, errc, quitC)
+			// doneC is nil, the error happened which is sent to errc channel, already
+			if wsDoneC == nil {
+				continue
+			}
+			go func() {
```

Comment:
dont we need to `rpcSubscriptionsWg.Add(1) ` here?

Reply:
Yes, I missed that, and it worked as the done channel was never closed, because of errc send being blocked in watchSubscriptionEvents. The change https://github.com/ethersphere/go-ethereum/pull/385/commits/b3be75c537dbb9aa1729ee08abb9768c462c966c addresses that.